extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -41,47 +40,80 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
 
   private var initGeoJson: GeoJson? = null
   private var initData: String? = null
+  private var initDataId: String? = null
 
   private constructor(
     builder: Builder,
     geoJson: GeoJson?,
     data: String?,
+    dataId: String?,
   ) : this(builder) {
     this.initGeoJson = geoJson
     this.initData = data
+    this.initDataId = dataId
   }
 
-  private fun setGeoJson(geoJson: GeoJson) {
+  private fun setGeoJson(geoJson: GeoJson, dataId: String? = null) {
     workerHandler.removeCallbacksAndMessages(null)
     workerHandler.post {
-      delegate?.setStyleGeoJSONSourceData(sourceId, toGeoJsonData(geoJson)) ?: logW(
-        TAG,
-        "GeoJsonSource (id=$sourceId) was not able to set data" +
-          " with `feature()`, `featureCollection()` or `geometry()`" +
-          " as there is no Style object."
-      )
+      if (dataId != null) {
+        delegate?.setStyleGeoJSONSourceData(
+          /* sourceId = */ sourceId,
+          /* dataId = */ dataId,
+          /* data = */ toGeoJsonData(geoJson)
+        ) ?: logW(
+          TAG,
+          "GeoJsonSource (id=$sourceId) was not able to set data" +
```

Comment:
1. Should we use better name than just `id=`?
2. Might be useful to log the `dataId`?

Reply:
Well, it's merely a warning message but I don't have objections to reword, will use `sourceId = ..., dataId = ...`